purchase_order_revision/models/purchase_order.py

```diff
@@ -30,7 +30,7 @@ def _prepare_revision_data(self, new_revision):
 
     def action_view_revisions(self):
         self.ensure_one()
-        action = self.env.ref("purchase.purchase_rfq")
+        action = self.env.ref("purchase.purchase_rfq").sudo()
         result = action.read()[0]
```

Comment:
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("purchase.purchase_rfq")
```
and replace `result` with `action` in the following lines.

Find the relevant comment in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#tasks-to-do-in-the-migration.

Relay the same update in the OCA repo.

Reply:
I have finished fixing the code and the functional test.